### PR TITLE
Falkonai/sr/extend outreach connector

### PR DESCRIPTION
**Pull request by rokadias**
Adding events and specific templates streams.

Along with pulling forward some changes from the new version of the connector to make it work.

### CODE REVIEW

File: airbyte-integrations/connectors/source-outreach/source_outreach/source.py [hunk 1]
```diff
@@ -22,21 +25,39 @@
 class OutreachStream(HttpStream, ABC):
 
     url_base = _URL_BASE
+    primary_key = "id"
+    page_size = 1000
```
Comment (joshzana) [inline airbyte-integrations/connectors/source-outreach/source_outreach/source.py:30]: Should this be in the config as well?  I would assume that a higher page size means fewer API calls, so there is some connection to `max_api_requests`

File: airbyte-integrations/connectors/source-outreach/source_outreach/source.py [hunk 2]
```diff
@@ -47,39 +68,89 @@ def next_page_token(self, response: requests.Response) -> Optional[Mapping[str,
...
+                **{self.primary_key: element[self.primary_key], **relationships},
+            }
 
 
 # Basic incremental stream
 class IncrementalOutreachStream(OutreachStream, ABC):
     @property
     def cursor_field(self) -> str:
-        return "attributes/properties/updatedAt"
+        return "updatedAt"
```
Comment (joshzana) [inline airbyte-integrations/connectors/source-outreach/source_outreach/source.py:124]: Thoughts on naming this `UpdatedAtIncrementalOutreachStream` for consistency?

File: airbyte-integrations/connectors/source-outreach/source_outreach/source.py [hunk 3]
```diff
@@ -121,34 +188,268 @@ def path(self, **kwargs) -> str:
...
+class DependentOutreachStream(OutreachStream, ABC):
+    def __init__(
+        self,
+        **kwargs,
+    ):
+        self.dependent_api_response_json = None
+        self.remaining_ids = None
+        super().__init__(**kwargs)
+
+    def _fill_out_remaining_ids(self) -> None:
```
Comment (joshzana) [inline airbyte-integrations/connectors/source-outreach/source_outreach/source.py:321]: Can you add some comments to this function?  Hard for me to understand what's going on.

**Review by joshzana — APPROVED** [review]

File: airbyte-integrations/connectors/source-outreach/source_outreach/source.py (hunk 1 above)
Comment (rokadias) [reply airbyte-integrations/connectors/source-outreach/source_outreach/source.py:30]: 1000 is the maximum page_size and no reason not to use the maximum afaict.

This was also to make it match what the new version of the connector looks like:
https://github.com/airbytehq/airbyte/blob/master/airbyte-integrations/connectors/source-outreach/source_outreach/source.py#L26

File: airbyte-integrations/connectors/source-outreach/source_outreach/source.py (hunk 2 above)
Comment (rokadias) [reply airbyte-integrations/connectors/source-outreach/source_outreach/source.py:124]: I wanted to, but figured for when merging back to the original. Looking at this implementation this seems like written by a beginner with airbyte integrations. :shrug:  

Current Version naming:
https://github.com/airbytehq/airbyte/blob/master/airbyte-integrations/connectors/source-outreach/source_outreach/source.py#L80

File: airbyte-integrations/connectors/source-outreach/source_outreach/source.py (hunk 3 above)
Comment (rokadias) [reply airbyte-integrations/connectors/source-outreach/source_outreach/source.py:321]: Added documentation, but yeah this area is a lot of haxing to work around a bug in Outreach's code, and not designed for how airbyte like's to consume streams.

**Pull request merged**